From: Oleksij Rempel <o.rempel@pengutronix.de>
Date: Thu, 5 Oct 2023 10:10:39 +0200
Subject: [PATCH] WIP: regulator: pca9450: add vin-supply DT support

For now, hard-code vin supplier to tell regulator framework that we
use vin-supply DT properties.

Properly we should probably walk over DT subnodes and verify which
actually have vin-supply property.

Signed-off-by: Oleksij Rempel <o.rempel@pengutronix.de>
---
 drivers/regulator/pca9450-regulator.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/regulator/pca9450-regulator.c b/drivers/regulator/pca9450-regulator.c
index f539ed5f1d82..12a10e77798f 100644
--- a/drivers/regulator/pca9450-regulator.c
+++ b/drivers/regulator/pca9450-regulator.c
@@ -481,6 +481,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.n_ramp_values = ARRAY_SIZE(pca9450_dvs_buck_ramp_table),
 			.owner = THIS_MODULE,
 			.of_parse_cb = pca9450_set_dvs_levels,
+			.supply_name = "vin",
 		},
 		.dvs = {
 			.run_reg = PCA9450_REG_BUCK1OUT_DVS0,
@@ -510,6 +511,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.n_ramp_values = ARRAY_SIZE(pca9450_dvs_buck_ramp_table),
 			.owner = THIS_MODULE,
 			.of_parse_cb = pca9450_set_dvs_levels,
+			.supply_name = "vin",
 		},
 		.dvs = {
 			.run_reg = PCA9450_REG_BUCK2OUT_DVS0,
@@ -534,6 +536,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_BUCK4CTRL,
 			.enable_mask = BUCK4_ENMODE_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 	{
@@ -552,6 +555,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_BUCK5CTRL,
 			.enable_mask = BUCK5_ENMODE_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 	{
@@ -570,6 +574,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_BUCK6CTRL,
 			.enable_mask = BUCK6_ENMODE_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 	{
@@ -588,6 +593,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_LDO1CTRL,
 			.enable_mask = LDO1_EN_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 	{
@@ -606,6 +612,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_LDO2CTRL,
 			.enable_mask = LDO2_EN_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 	{
@@ -624,6 +631,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_LDO3CTRL,
 			.enable_mask = LDO3_EN_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 	{
@@ -642,6 +650,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_LDO4CTRL,
 			.enable_mask = LDO4_EN_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 	{
@@ -660,6 +669,7 @@ static const struct pca9450_regulator_desc pca9450bc_regulators[] = {
 			.enable_reg = PCA9450_REG_LDO5CTRL_H,
 			.enable_mask = LDO5H_EN_MASK,
 			.owner = THIS_MODULE,
+			.supply_name = "vin",
 		},
 	},
 };
